corehq/apps/app_manager/views/view_generic.py

```diff
@@ -80,7 +80,7 @@ def view_generic(request, domain, app_id=None, module_id=None, form_id=None,
                 raise Http404()
             module_id = module.id
 
-        if form_id:
+        if form_id and module is not None:
```

Comment:
This (form_id but no module) should never happen, right? Might be preferable to just fail hard here on the undefined module rather than having unexpected behavior (a redirect to the wrong page?) later on.